pandas/tests/io/xml/test_xml.py

```diff
@@ -1351,18 +1351,66 @@ def test_string_error(parser):
         )
 
 
-def test_file_like_error(datapath, parser, mode):
+def test_file_like_iterparse(datapath, parser, mode):
     filename = datapath("io", "data", "xml", "books.xml")
-    with pytest.raises(
-        ParserError, match=("iterparse is designed for large XML files")
-    ):
-        with open(filename) as f:
-            read_xml(
+
+    with open(filename, mode) as f:
+        if mode == "r" and parser == "lxml":
+            with pytest.raises(
+                TypeError, match=("reading file objects must return bytes objects")
+            ):
+                read_xml(
+                    f,
+                    parser=parser,
+                    iterparse={
+                        "book": ["category", "title", "year", "author", "price"]
+                    },
+                )
+            return None
+        else:
+            df_filelike = read_xml(
                 f,
                 parser=parser,
                 iterparse={"book": ["category", "title", "year", "author", "price"]},
             )
 
+    IOfunc = StringIO if mode == "r" else BytesIO
+    with open(filename, mode) as f:
```

Comment:
Could you make this new block a separate test?

Reply:
Sounds good. Will do.